### ISSUE TITLE
False positive: local scope static variables are mislabled as undefined when accessed

### ISSUE THREAD

**Issue by ikari7789** [phpmd/phpmd#912]
- PHPMD version: 2.10.2
- PHP Version: 7.2.34
- Installation type: composer
- Operating System / Distribution & Version: macOS 11.6

### Current Behavior

The clean code `UndefinedVariable` rule reports a violation on local scope static variables.

```
FILE: /path/to/example.php
--------------------------------------------------------------------------------------------------------
 7   | VIOLATION | Avoid using undefined variables such as '$variable' which will lead to PHP notices.
```

### Expected Behavior

Variable should be seen as defined and not report any violations.

### Steps To Reproduce:

```xml
<?xml version="1.0"?>
<ruleset name="PHP Mess Detector Ruleset"
    xmlns="http://pmd.sf.net/ruleset/1.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0 http://pmd.sf.net/ruleset_xml_schema.xsd"
    xsi:noNamespaceSchemaLocation="http://pmd.sf.net/ruleset_xml_schema.xsd">

    <rule ref="rulesets/cleancode.xml/UndefinedVariable" />
</ruleset>
```

```php
<?php

function staticVariableUndefined()
{
    static $variable = 0;

    ++$variable;
}
```

```shell
phpmd example.php ansi example.xml
```

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by AJenbo** [comment]
Fixed in 3.x
